src/Plugin/AccessControlHierarchy/Taxonomy.php

```diff
@@ -328,7 +329,11 @@ public function buildConfigurationForm(array $form, FormStateInterface $form_sta
             $handler_settings = $field_definitions[$field_name]->getSetting('handler_settings');
             // Must refer to a proper target. Target bundles referring to
             // themselves would create an infinite loop. Deny.
-            if (!isset($handler_settings['target_bundles']) || in_array($bundle, $handler_settings['target_bundles'], TRUE)) {
+            if ($entity_type_id == 'taxonomy_term' && in_array($bundle, $this->configuration['vocabularies'])) {
+              continue;
+            }
+            $allowed = array_intersect($handler_settings['target_bundles'], $this->configuration['vocabularies']);
+            if (!isset($handler_settings['target_bundles']) || empty($allowed)) {
```

Comment:
we've already accessed $handler_settings['target_bundles'] without the isset check. So either its always set (and we don't need it) or we risk triggering a warning here by accessing it without checking first.